ASP/ASP.sublime-syntax

```diff
@@ -8,25 +8,49 @@ file_extensions:
 scope: source.asp
 contexts:
   main:
-    - match: '^\s*((?i:function|sub))\s*([a-zA-Z_]\w*)\s*(\()([^)]*)(\)).*\n?'
-      scope: meta.function.asp
+    - match: '^\s*((?i:function|sub))\s*'
       captures:
         1: storage.type.function.asp
-        2: entity.name.function.asp
-        3: punctuation.definition.parameters.asp
-        4: variable.parameter.function.asp
-        5: punctuation.definition.parameters.asp
-    - match: (').*$\n?
-      scope: comment.line.apostrophe.asp
-      captures:
-        1: punctuation.definition.comment.asp
-    - match: (REM ).*$\n?
-      scope: comment.line.rem.asp
-      captures:
-        1: punctuation.definition.comment.asp
+      push:
+        - meta_scope: meta.function.asp
+        - include: span_mulitple_lines
+        - match: '[a-zA-Z_]\w*'
+          scope: entity.name.function.asp
+          set:
+            - match: '\s+'
+              scope: meta.function.asp
+            - match: '(?=[_(])'
+              set:
+                - meta_content_scope: meta.function.asp
+                - include: span_mulitple_lines
+                - match: \(
+                  scope: punctuation.definition.parameters.asp
+                  set:
+                    - meta_content_scope: meta.function.asp
+                    - include: span_mulitple_lines
+                    - match: \)
+                      scope: punctuation.definition.parameters.asp
+                      set:
+                        - include: span_mulitple_lines
+                        - match: '\S+'
+                          scope: invalid.illegal.unexpected-token-after-method-declaration.asp
+                        - match: '$'
+                          pop: true
+                    - match: .
+                      scope: variable.parameter.function.asp
+            - include: comments
+            - match: \s*\S+.*$
+              scope: invalid.illegal.unexpected-token-after-method-declaration.asp
+              pop: true
+            - match: ''
+              pop: true
+    - include: comments
     - match: (?i:\b(If|Then|Else|ElseIf|End If|While|Wend|For|To|Each|In|Step|Case|Select|End Select|Return|Continue|Do|Until|Loop|Next|With|Exit Do|Exit For|Exit Function|Exit Property|Exit Sub)\b)
       scope: keyword.control.asp
-    - match: '=|>=|<|>|<|<>|\+|-|\*|\^|&|\b(?i:(Mod|And|Not|Or|Xor|Is))\b'
+    - match: '&[hH][0-9a-fA-F]+&?|\b(((([0-9]+\.?[0-9]*)|(\.[0-9]+))((e|E)(\+|-)?[0-9]+)?)(L|l|UL|ul|u|U|F|f)?)\b'
```

Comment:
for (my/future reader) info: `\h` is specific to the Ruby regex flavor. In other flavors (Perl, PCRE etc.), it matches horizontal whitespace. http://www.regular-expressions.info/shorthand.html